GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+
+                ITenantDetail tenantDetail = await GetTenantDetails(client);
+                User signedInUser = await GetSignedInUser(client);
+                await UpdateSignedInUsersPhoto(signedInUser);
+
+                Console.WriteLine("\nSearching for any user based on UPN, DisplayName, First or Last Name");
+                Console.WriteLine("\nPlease enter the user's name you are looking for:");
+                string searchString = Console.ReadLine();
+                await PeoplePickerExample(client, searchString);
+
+                await PrintUsersManager(signedInUser);
+                newUser = await CreateNewUser(client,
+                    tenantDetail.VerifiedDomains.First(x => x.@default.HasValue && x.@default.Value));
+                await UpdateNewUser(newUser);
+                await ResetUserPassword(newUser);
+                await OtherUserWriteOperations(client, newUser, signedInUser);
+
+                newGroup = await CreateNewGroup(client);
+                await PrintAllRoles(client, searchString);
+                await PrintServicePrincipals(client);
+
+                await PrintApplications(client);
+                newApp = await CreateNewApplication(client, newUser);
+                ServicePrincipal newServicePrincipal = await CreateServicePrincipal(client, newApp);
+                string extName = "linkedInUserId";
+                await CreateSchemaExtensions(client, newApp, extName);
+                await ManipulateExtensionProperty(newApp, extName, newUser);
+                PrintExtensionProperty(newUser, extName);
+                await AssignAppRole(client, newApp, newServicePrincipal);
+
+                await PrintDevices(client);
+                await CreateOAuth2Permission(client, newServicePrincipal);
+                await PrintAllPermissions(client);
+
+                await PrintAllDomains(client);
+                newDomain = await CreateNewDomain(client);
+
+                await BatchOps(client);
+            }
+            finally
+            {
+                DeleteUser(newUser).Wait();
+                DeleteApplication(newApp).Wait();
+                DeleteDomain(newDomain).Wait();
+                DeleteGroup(newGroup).Wait();
+            }
+        }
+
+        public static async Task AppMode()
+        {
+            #region Setup Microsoft Graph client for app
+
+            ActiveDirectoryClient client;
+            //*********************************************************************
+            // setup Microsoft Graph client for app
+            //*********************************************************************
+            try
+            {
+                client = AuthenticationHelper.GetActiveDirectoryClientAsApplication();
+            }
+            catch (Exception ex)
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("Acquiring a token failed with the following error: {0}", ex.Message);
```

Comment:
Done.

---
In reply to: [80758343](https://github.com/Azure-Samples/active-directory-dotnet-graphapi-console/pull/37#discussion_r80758343) [](ancestors = 80758343)